src/io/inputs/model-converter/convertorVisitor.cpp

```diff
@@ -389,32 +397,137 @@ std::any ConvertorVisitor::visitDual(ExprParser::DualContext* context)
         return node;
     }
 
-    throw NoConstraintWithThisName(constraint_id);
+    throw NoConstraintWithThisName(model_.id, constraint_id);
 }
 
-std::any ConvertorVisitor::visitReducedCost(ExprParser::ReducedCostContext* context)
+std::any ConvertorVisitor::handleReducedCost(ExprParser::ArgListContext* context)
 {
-    const auto& variables = model_.variables;
-    for (std::size_t index = 0; index < variables.size(); ++index)
+    std::vector<Node*> nodes;
+    try
     {
-        const auto& var = variables[index];
-        if (var.id == context->IDENTIFIER()->getText())
-        {
-            return static_cast<Node*>(
-              registry_.create<ReducedCostNode>(var.id,
-                                                index,
-                                                convertToTimeIndex(var.time_dependent,
-                                                                   var.scenario_dependent)));
-        }
+        nodes = std::any_cast<std::vector<Node*>>(context->accept(this));
+    }
+    catch (NoParameterOrVariableWithThisName) // to print accurate message
+    {
+        throw NoVariableWithThisName(model_.id, context->expr(0)->getText());
     }
+    if (nodes.size() != 1)
+    {
+        throw std::invalid_argument("reduced_cost operator expect only one variable id");
+    }
+
+    return static_cast<Node*>(
+      registry_.create<FunctionNode>(FunctionNodeType::reduced_cost, nodes.at(0)));
+}
+template<class T>
+std::any ConvertorVisitor::processPower(std::vector<T*> exprContexts,
+                                        const std::function<std::string()>& toStringTreeCallBack)
+{
+    if (exprContexts.size() != 2)
+    {
+        throw std::invalid_argument("power operator expect only two arguments got "
+                                    + std::to_string(exprContexts.size()) + " in "
+                                    + toStringTreeCallBack());
+    }
+    if (exprContexts.at(0) == nullptr)
+    {
+        throw std::invalid_argument("bad power expression, the base is invalid in "
+                                    + toStringTreeCallBack());
+    }
+    if (exprContexts.at(1) == nullptr)
+    {
+        throw std::invalid_argument("bad power expression, the exponent is invalid in "
+                                    + toStringTreeCallBack());
+    }
```

Comment:
Not sure this can happen

Reply:
Let me Check. However this is helpful in case of syntaxe error, could be used elsewhere